pkg/providers/instance/aksmachineinstance.go

```diff
@@ -0,0 +1,580 @@
+			aksMachineID,
+			aksMachineNodeImageVersion,
+			vmResourceID,
+		), nil
+	} else if !IsARMNotFound(err) {
+		// Not fatal. Will fall back to normal creation.
+		log.FromContext(ctx).Error(err, "failed to check for existing machine", "aksMachineName", aksMachineName)
+	}
+
+	// Decide on offerings
+	instanceType, capacityType, zone := offerings.PickSkuSizePriorityAndZone(ctx, nodeClaim, instanceTypes)
+	if instanceType == nil {
+		return nil, corecloudprovider.NewInsufficientCapacityError(fmt.Errorf("no instance types available"))
+	}
+
+	// Build the AKS machine template
+	aksMachineTemplate, err := p.buildAKSMachineTemplate(ctx, instanceType, capacityType, zone, nodeClass, nodeClaim)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build AKS machine template from template: %w", err)
+	}
+	log.FromContext(ctx).V(1).Info("XPMT: AKS machine template built", "machineObj", *aksMachineTemplate)
+
+	// Call the AKS machine API with the template to create the AKS machine instance
+	log.FromContext(ctx).V(1).Info("creating AKS machine instance", "aksMachineName", aksMachineName, "instance-type", instanceType.Name)
+
+	// Resolve VM image ID
+	// E.g., "/subscriptions/10945678-1234-1234-1234-123456789012/resourceGroups/AKS-Ubuntu/providers/Microsoft.Compute/galleries/AKSUbuntu/images/2204gen2containerd/versions/2022.10.03"
+	vmImageID, err := p.imageResolver.ResolveNodeImageFromNodeClass(nodeClass, instanceType)
+	if err != nil {
+		return nil, fmt.Errorf("failed to resolve VM image ID: %w", err)
+	}
+	osImageSubscriptionID, osImageResourceGroup, osImageGallery, osImageName, osImageVersion, err := parseVMImageID(vmImageID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse VM image ID %q: %w", vmImageID, err)
+	}
+	header := http.Header{}
+	header.Set("AKSHTTPCustomFeatures", "Microsoft.ContainerService/UseCustomizedOSImage")
+	header.Set("OSImageName", osImageName)                          // E.g. "2204gen2containerd"
+	header.Set("OSImageResourceGroup", osImageResourceGroup)        // E.g. "AKS-Ubuntu"
+	header.Set("OSImageSubscriptionID", osImageSubscriptionID)      // E.g., "10945678-1234-1234-1234-123456789012"
+	header.Set("OSImageGallery", osImageGallery)                    // E.g., "AKSUbuntu"
+	header.Set("OSImageVersion", osImageVersion)                    // E.g., "2022.10.03"
+	ctxWithHeader := context.WithValue(ctx, "vmimageid", vmImageID) // This line is really just for testing (see fake/aksmachinesapi.go). Azure-sdk-for-go is restrictive in extracting the header out.
+	ctxWithHeader = policy.WithHTTPHeader(ctxWithHeader, header)
+
+	poller, err := p.azClient.aksMachinesClient.BeginCreateOrUpdate(ctxWithHeader, p.clusterResourceGroup, p.clusterName, p.aksMachinesPoolName, aksMachineName, *aksMachineTemplate, nil)
+	if err != nil {
+		return nil, fmt.Errorf("aksMachine.BeginCreateOrUpdate for AKS machine %q failed: %w", aksMachineName, err)
+	}
+
+	// Get once after begin create to retrieve VMResourceID.
```

Comment:
minor: Worth a TODO that we can probably extract this from response to BeginCreateOrUpdate anbd about second GET if we need to